src/users/entities/users.repository.ts

```diff
@@ -40,6 +41,39 @@ export class UsersRepository {
     return this.repo.find();
   }
 
+  queryBuilderWithFilters(filters: UsersFilters): SelectQueryBuilder<User> {
+    const qb = this.repo.createQueryBuilder('people');
+
+    qb.innerJoinAndSelect('people.organization', 'organization');
+    const { firstName, lastName, email, organization, role } = filters;
+
+    if (firstName) {
+      qb.andWhere('people.firstName LIKE :firstName', {
+        firstName: `${firstName}%`,
+      });
+    }
+
+    if (lastName) {
+      qb.andWhere('people.lastName LIKE :lastName', {
+        lastName: `${lastName}%`,
+      });
+    }
+
+    if (email) {
+      qb.andWhere('people.email LIKE :email', { email: `${email}%` });
+    }
+
+    if (organization) {
+      qb.andWhere('organization.id = :organization', { organization });
+    }
+
+    if (role) {
+      qb.andWhere('people.roles = :role', { role });
```

Comment:
This filter won't work like that. It needs to use a JSON function to match that the requested role is contained in the JSON field in the database.